www/js/app.js

```diff
@@ -1627,9 +1627,9 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
                 alert("Error finding random article.");
             } else {
                 // We fall back to the old A namespace to support old ZIM files without a text/html MIME type for articles
-                // DEV: This will need to be changed if we search titlePtrList version 1
-                // in a future PR, as that list contains only articles
-                if (dirEntry.getMimetype() === 'text/html' || dirEntry.namespace === 'A') {
+                // DEV: If articlePtrPos is defined in zimFile, then we are using a v1 article-only title listing. By definition,
+                // all dirEntries in an article-only listing must be articles.  
+                if (selectedArchive._file.articlePtrPos || dirEntry.getMimetype() === 'text/html' || dirEntry.namespace === 'A') {
```

Comment:
NB With the currently available test ZIMs that have empty v1 title pointer lists, `articlePtrPos` will point to a v0 title pointer list instead of a v1. This is so we can test. However, the side effect is that this random function will incorrectly show non-article dirEntries when used on one of these test ZIMs. As soon as we have a valid test ZIM, this should start to work correctly.